module/x/gravity/keeper/grpc_query.go

```diff
@@ -132,6 +132,8 @@ func (k Keeper) BatchTxs(c context.Context, req *types.BatchTxsRequest) (*types.
 		return nil, err
 	}
 
+	batches = orderBatchesByNonceAscending(batches)
```

Comment:
Should we put this ordering in the keeper function, in the same way unsigned batches are handled?

Reply:
Oh this made me realize how weird it is to try to order pagination results anyway. Does order by nonce really mean anything if the pages are selected based on order of contract address?

Reply:
I think we'd want to have the source data for the pages already be sorted appropriately.

Reply:
Here the source data is going to be sorted by target contract address first because of how the batch key is structured.

Reply:
I mean, we're implementing `PaginateOutgoingTxsByType`, right? We can have it handle the sort order internally and output paginated data on the correct ordering.